lib/logproto/logproto-buffered-server.c

```diff
@@ -415,6 +415,58 @@ log_proto_buffered_server_convert_state(LogProtoBufferedServer *self, guint8 per
   return FALSE;
 }
 
+static void
+_log_proto_buffered_server_swap_bytes_if_needed(PersistState *persist_state, PersistEntryHandle old_state_handle)
+{
+  LogProtoBufferedServerState *state;
+  state = (LogProtoBufferedServerState*)(persist_state_map_entry(persist_state, old_state_handle));
+
+  if ((state->header.big_endian && G_BYTE_ORDER == G_LITTLE_ENDIAN) ||
+      (!state->header.big_endian && G_BYTE_ORDER == G_BIG_ENDIAN))
+    {
+
+      /* byte order conversion in order to avoid the hassle with
+          scattered byte order conversions in the code */
+
+      state->header.big_endian = !state->header.big_endian;
+      state->buffer_pos = GUINT32_SWAP_LE_BE(state->buffer_pos);
+      state->pending_buffer_pos = GUINT32_SWAP_LE_BE(state->pending_buffer_pos);
+      state->pending_buffer_end = GUINT32_SWAP_LE_BE(state->pending_buffer_end);
+      state->buffer_size = GUINT32_SWAP_LE_BE(state->buffer_size);
+      state->raw_stream_pos = GUINT64_SWAP_LE_BE(state->raw_stream_pos);
+      state->raw_buffer_size = GUINT32_SWAP_LE_BE(state->raw_buffer_size);
+      state->pending_raw_stream_pos = GUINT64_SWAP_LE_BE(state->pending_raw_stream_pos);
+      state->pending_raw_buffer_size = GUINT32_SWAP_LE_BE(state->pending_raw_buffer_size);
+      state->file_size = GUINT64_SWAP_LE_BE(state->file_size);
+      state->file_inode = GUINT64_SWAP_LE_BE(state->file_inode);
+    }
+
+  persist_state_unmap_entry(persist_state, old_state_handle);
+}
+
+static gboolean
+_log_proto_buffered_server_convert_header_version(PersistState *persist_state, PersistEntryHandle old_state_handle)
+{
+  LogProtoBufferedServerState *state;
+  gboolean result = TRUE;
+
+  state = persist_state_map_entry(persist_state, old_state_handle);
+  if (state->header.version > 1)
+    {
+      msg_error("Internal error restoring log reader state, stored data is too new",
+                evt_tag_int("version", state->header.version));
+      result = FALSE;
+    }
+
+  if (state->header.version == 0)
+    {
```

Comment:
Is something missing here? As I see, if version is zero, we assign zero to the version, which is a noop.